### PR TITLE
Initial File Structure and Beginning of Application List

### LINKED ISSUE #6: Application List
Applications need to be listed in the app, in order. Should be built for extensive customization.

## Task
Build an "app list" in React Native for the front page of the authenticator app. Each list item should contain the following:

**List Item**
- [ ] Application Name [Text]
- [ ] Application Icon [Image]
- [ ] Username for account [Text]
- [ ] TOTP password [Text]

### PR DESCRIPTION
**Pull request by jmabramo**

### CODE REVIEW

File: authenticator/components/application-list-item.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        iconSrc: '',
+        totp: '000000'
+    },
+    '000001': {
+        name: 'App 2',
+        username: 'imanotheruser',
+        iconSrc: '',
+        totp: '111111'
+    }
+}
```
Comment (emilkovacev) [inline authenticator/components/application-list-item.js]: We should abstract this data storage out of the application before we merge into main. In the meantime, we can just add a comment that it needs to be done and leave it for another PR

File: authenticator/components/application-list-totp.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import { StyleSheet, Text, View } from 'react-native';
+
+export default function ApplicationListTOTP( id ) {
+    return(
+        <View>
+            <Text style={styles.totpText}>000000</Text>
+        </View>
+    );
+};
```
Comment (emilkovacev) [inline authenticator/components/application-list-totp.js]: this is simple enough to add straight to the Application List Icon component instead of making an entirely new component. Especially since this seems like the only time we're using an Application List TOTP (in the front-end)

File: authenticator/components/application-list-icon.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+        </View>
+    );
+}
+
+const styles = StyleSheet.create({
+    logo: {
+      width: 75,
+      height: 75,
+    },
+  });
```
Comment (emilkovacev) [inline authenticator/components/application-list-icon.js]: this is simple enough to add directly to the application list item component

Comment (emilkovacev) [reply authenticator/components/application-list-icon.js]: I can see the advantages of abstracting this to a component though, especially if we're using Icons throughout the website, so it definitely could work to our advantage in the long run. Let me know what you think about this!

File: authenticator/babel.config.js [hunk 4]
```diff
@@ -0,0 +1,6 @@
+module.exports = function(api) {
+  api.cache(true);
+  return {
+    presets: ['babel-preset-expo'],
```
Comment (emilkovacev) [inline authenticator/babel.config.js:4]: I looked this up, and this seems like a really great addition to presets :)

File: authenticator/components/application-list-icon.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+        <View>
+            <Image style={styles.logo} source={{uri: 'https://reactnative.dev/img/tiny_logo.png'}}/>
+        </View>
+    );
+}
+
+const styles = StyleSheet.create({
+    logo: {
+      width: 75,
+      height: 75,
```
Comment (emilkovacev) [inline authenticator/components/application-list-icon.js]: can we set the width and height as an input for the Application list view? If we want to use an icon or allow for better readability that would make it super easy

File: authenticator/App.js [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+      <ApplicationList></ApplicationList>
+      <StatusBar style="auto" />
+    </View>
+  );
+}
+
+const styles = StyleSheet.create({
+  container: {
+    flex: 1,
+    backgroundColor: '#fff',
```
Comment (emilkovacev) [inline authenticator/App.js:29]: Can we add background color as a parameter to the App container?

File: authenticator/components/application-list-item.js [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+        alignItems: 'center',
+        margin: '10px',
+        height: '100px',
+    },
+    nameText: {
+        fontSize:20
+    },
+    usernameText:{
+        fontStyle: 'italic'
+    }
```
Comment (emilkovacev) [inline authenticator/components/application-list-item.js]: Can we add fontSize (and maybe fontStyle) as variables?

File: authenticator/views/application-list.js [hunk 8]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import { StyleSheet, View, FlatList } from 'react-native';
+import ApplicationListItem from "../components/application-list-item"
+
+let example_data =
+    [
+        '000000',
+        '000001',
+        
+    ];
```
Comment (emilkovacev) [inline authenticator/views/application-list.js]: We should abstract this out and put a comment so we know there needs to be permanent data storage linked here

**Review by emilkovacev — CHANGES_REQUESTED** [review]

**Comment by emilkovacev** [comment]
This is a really awesome pull request! Don't be deceived by the number of comments I made, most of this looks good to go! We can make a couple of small changes to simplify the implementation structure, but they shouldn't take too long. Great job! 😄

File: authenticator/babel.config.js (hunk 4 above)
Comment (jmabramo) [reply authenticator/babel.config.js:4]: That was part of expo's default blank project initializer

File: authenticator/components/application-list-icon.js (hunk 3 above)
Comment (jmabramo) [reply authenticator/components/application-list-icon.js]: The main reason I separated it here was I assumed that we would need a way to get an icon.  My idea was to somehow use the favicon from the site (similar to the way Bitwarden password manager gets icons).  I thought the logic for that might be better abstracted away in this file eventually.

File: authenticator/views/application-list.js (hunk 8 above)
Comment (jmabramo) [reply authenticator/views/application-list.js]: It's only there for visualization purposes because there is no backend yet.

File: authenticator/components/application-list-icon.js (hunk 3 above)
Comment (emilkovacev) [reply authenticator/components/application-list-icon.js]: Ah ok that makes a lot of sense, let's go with that

File: authenticator/views/application-list.js (hunk 8 above)
Comment (emilkovacev) [reply authenticator/views/application-list.js]: That makes sense, I think we should remove it before merging to main

**Review by emilkovacev — APPROVED** [review]

**Review by ahanabhattchrya — APPROVED** [review]
